ChatSecure/Classes/Controllers/XMPP/OTRYapDatabaseRosterStorage.m

```diff
@@ -180,6 +180,11 @@ - (void)updateBuddy:(nullable OTRXMPPBuddy *)buddy withItem:(nonnull NSXMLElemen
     [[OTRDatabaseManager sharedInstance].readWriteDatabaseConnection asyncReadWriteWithBlock:^(YapDatabaseReadWriteTransaction *transaction) {
         [newBuddy saveWithTransaction:transaction];
     }];
+    
+    if (buddy.pendingApproval && !newBuddy.pendingApproval) {
+        // Buddy was approved
+        [[NSNotificationCenter defaultCenter] postNotificationName:OTRBuddyPendingApprovalDidChangeNotification object:buddy userInfo:nil];
```

Comment:
Ok, makes sense, we should probably change the calls in OTREncryptionManager.m as well then, I was kind of inspired by that =)